apps/utils/deletion.py

```diff
@@ -1,8 +1,81 @@
+from collections import Counter
+from functools import reduce
+from operator import or_
 from typing import Any
 
 from django.contrib.admin.utils import NestedObjects
-from django.db import router
+from django.db import models, router, transaction
 from django.db.models.deletion import get_candidate_relations_to_delete
+from field_audit.field_audit import get_audited_models
+
+
+def delete_object_with_auditing_of_related_objects(obj):
+    """Deletes the given object and its related objects, auditing the deletion of each object.
+
+    Args:
+        obj: The object to delete.
+    """
+    from field_audit.models import AuditAction
+
+    collector = NestedObjects(using="default")
+    collector.collect([obj])
+
+    models_to_delete = set(collector.data)
+
+    updates_not_part_of_delete = {}
+    if collector.field_updates:
+        for (field, value), instances_list in collector.field_updates.items():
+            model = field.model
+            if model in models_to_delete or model._meta.auto_created or model not in get_audited_models():
+                continue
+            updates_not_part_of_delete[(field, value)] = instances_list
+
+    counter = Counter()
+    with transaction.atomic():
+        _perform_updates_for_delete(updates_not_part_of_delete)
+
+        for model, instances in reversed(collector.data.items()):
+            if model._meta.auto_created:
```

Comment:
Why not delete these as well?

Reply:
Django will handle their deletion itself when the models they refer to are deleted.